Cubical/HITs/S1/Base.agda

```diff
@@ -90,6 +92,159 @@ windingIntLoop (negsuc (suc n)) = λ i → predInt (windingIntLoop (negsuc n) i)
 ΩS¹≡Int : ΩS¹ ≡ Int
 ΩS¹≡Int = isoToPath (iso winding (decode base) windingIntLoop (decodeEncode base))
 
+-- intLoop and winding are group homomorphisms
+
+intLoop-sucInt : (z : Int) → intLoop (sucInt z) ≡ (intLoop z) · loop
+intLoop-sucInt (pos n)          = refl
+intLoop-sucInt (negsuc zero)    = sym (compPath-inv-l loop)
+intLoop-sucInt (negsuc (suc n)) =
+  (sym (compPath-refl-r (intLoop (negsuc n))))
+  · (λ i → intLoop (negsuc n) · (compPath-inv-l loop (~ i)))
+  · (sym (compPath-assoc (intLoop (negsuc n)) (sym loop) loop))
+
+intLoop-predInt : (z : Int) → intLoop (predInt z) ≡ (intLoop z) · (sym loop)
+intLoop-predInt (pos zero)    = sym (compPath-refl-l (sym loop))
+intLoop-predInt (pos (suc n)) =
+  (sym (compPath-refl-r (intLoop (pos n))))
+  · (λ i → intLoop (pos n) · (compPath-inv-r loop (~ i)))
+  · (sym (compPath-assoc (intLoop (pos n)) loop (sym loop)))
+intLoop-predInt (negsuc n)    = refl
+
+intLoop-hom : (a : Int) → (b : Int) → (intLoop a) · (intLoop b) ≡ intLoop (a + b)
+intLoop-hom a (pos zero)       = compPath-refl-r (intLoop a)
+intLoop-hom a (pos (suc n))    =
+  (sym (compPath-assoc (intLoop a) (intLoop (pos n)) loop))
+  · (λ i → (intLoop-hom a (pos n) i) · loop)
+  · (sym (intLoop-sucInt (a + pos n)))
+intLoop-hom a (negsuc zero)    = sym (intLoop-predInt a)
+intLoop-hom a (negsuc (suc n)) =
+  (sym (compPath-assoc (intLoop a) (intLoop (negsuc n)) (sym loop)))
+  · (λ i → (intLoop-hom a (negsuc n) i) · (sym loop))
+  · (sym (intLoop-predInt (a + negsuc n)))
+
+winding-hom : (a : ΩS¹) → (b : ΩS¹) → winding (a · b) ≡ (winding a) + (winding b)
+winding-hom a b i =
+  hcomp (λ t → λ { (i = i0) → winding ((decodeEncode base a t) · (decodeEncode base b t))
+                 ; (i = i1) → windingIntLoop ((winding a) + (winding b)) t })
+        (winding (intLoop-hom (winding a) (winding b) i))
+
+-- Based homotopy group
+
+basedΩS¹ : (x : S¹) → Set
+basedΩS¹ x = x ≡ x
+
+-- Proof that the homotopy group is actually independent on the basepoint
```

Comment:
Like we discussed yesterday, can we change the encode-decode proof to prove this directly?

Reply:
I tried this myself and it doesn't directly work, so let's not worry about it!